inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,421 @@
+        }
+
+        project = getModel().getObject().getProject();
+        List<DocumentRepository> repositories = externalSearchService
+            .listDocumentRepositories(project);
+
+        documentImporter = new DocumentImporter(externalSearchService, documentService,
+            userRepository.getCurrentUser(), project);
+
+        ExternalSearchUserState searchState =
+            searchStateModel.getObject();
+        if (searchState.getCurrentRepository() == null && repositories.size() > 0) {
+            searchState.setCurrentRepository(repositories.get(0));
+        }
+        if (searchState.getTargetQuery() == null) {
+            searchState.setTargetQuery(Model.of(""));
+        }
+
+        repositoriesModel = LoadableDetachableModel.of(() -> externalSearchService
+            .listDocumentRepositories(project));
+
+        mainContainer = new WebMarkupContainer("mainContainer");
+        mainContainer.setOutputMarkupId(true);
+        add(mainContainer);
+
+        DocumentRepositorySelectionForm projectSelectionForm = new DocumentRepositorySelectionForm(
+            "repositorySelectionForm");
+        mainContainer.add(projectSelectionForm);
+
+        SearchForm searchForm = new SearchForm("searchForm");
+        add(searchForm);
+        mainContainer.add(searchForm);
+
+        List<IColumn<ExternalSearchResult, String>> columns = new ArrayList<>();
+
+        columns.add(new AbstractColumn<ExternalSearchResult, String>(new Model<>("Results"))
+        {
+            @Override
+            public void populateItem(Item<ICellPopulator<ExternalSearchResult>> cellItem,
+                String componentId, IModel<ExternalSearchResult> model)
+            {
+                @SuppressWarnings("rawtypes")
+                Item rowItem = cellItem.findParent( Item.class );
+                int rowIndex = rowItem.getIndex();
+                ResultRowView rowView = new ResultRowView(componentId, rowIndex + 1, model);
+                cellItem.add(rowView);
+            }
+        });
+
+        if (searchState.getDataProvider() == null)
+        {
+            searchState.setDataProvider(new ExternalResultDataProvider(externalSearchService,
+                userRepository.getCurrentUser(),
+                searchStateModel.getObject().getCurrentRepository(),
+                ""));
+        }
+
+        dataTableContainer = new WebMarkupContainer("dataTableContainer");
+        dataTableContainer.setOutputMarkupId(true);
+        mainContainer.add(dataTableContainer);
+
+        DataTable<ExternalSearchResult, String> resultTable = new DefaultDataTable<>("resultsTable",
+            columns, searchState.getDataProvider(), 8);
+        resultTable.setCurrentPage(searchState.getCurrentPage());
+        dataTableContainer.add(resultTable);
+
+    }
+
+    @Override protected void onDetach()
+    {
+        // TODO is it possible to know page change event and alter the state there?
```

Comment:
Does this TODO need to be addressed before merging?  Or does it need to become its own issue? When is `onDetach` called that differs from a page change

Reply:
This was an alternative I was thinking, to update the state of the selected search result page as it changes. As oppose to save the page number when switching to another sidebar. I removed the comment.